implementations/rust/ockam/ockam_command/src/tcp/inlet/create.rs

```diff
@@ -273,95 +277,160 @@ impl CreateCommand {
+        for to in cases {
+            CreateCommand::parse_arg_to(&state, to, None)
+                .await
+                .expect_err("Invalid protocol");
+        }
 
-        // The placeholders are replaced when using the arg's default value
-        let res = CreateCommand::parse_arg_to(&state, default_to_addr(), default_project_name)
+        // "to" default value
+        let res = CreateCommand::parse_arg_to(&state, default_to_addr(), None)
             .await
             .unwrap();
         assert_eq!(
             res,
-            "/project/p1/service/forward_to_default/secure/api/service/outlet"
+            "/project/p1/service/forward_to_default/secure/api/service/outlet".to_string()
         );
 
-        // The user provides a full project route
-        let addr = "/project/p1/service/forward_to_n1/secure/api/service/outlet";
-        let res = CreateCommand::parse_arg_to(&state, addr, default_project_name)
-            .await
-            .unwrap();
-        assert_eq!(res, addr);
+        // "to" argument accepts a full route
+        let cases = [
+            ("/project/p2/service/forward_to_n1/secure/api/service/myoutlet", None),
+            ("/worker/603b62d245c9119d584ba3d874eb8108/service/forward_to_n3/service/hop/service/outlet", None),
+            (&format!("/node/{node_name}/service/myoutlet"), Some(format!("/ip4/127.0.0.1/tcp/{node_port}/service/myoutlet"))),
+        ];
+        for (to, expected) in cases {
+            let res = CreateCommand::parse_arg_to(&state, to, None).await.unwrap();
+            let expected = expected.unwrap_or(to.to_string());
+            assert_eq!(res, expected);
+        }
 
-        // The user provides the name of the relay
-        let res = CreateCommand::parse_arg_to(&state, "alice", default_project_name)
+        // "to" argument accepts the name of the service
+        let res = CreateCommand::parse_arg_to(&state, "myoutlet", None)
             .await
             .unwrap();
         assert_eq!(
             res,
-            "/project/p1/service/forward_to_alice/secure/api/service/outlet"
+            "/project/p1/service/forward_to_default/secure/api/service/myoutlet".to_string()
         );
+
+        // "via" argument is used to replace the relay name
+        let cases = [
+            (
+                default_to_addr(),
+                "myrelay",
+                "/project/p1/service/forward_to_myrelay/secure/api/service/outlet",
+            ),
+            (
+                "myoutlet".to_string(),
+                "myrelay",
+                "/project/p1/service/forward_to_myrelay/secure/api/service/myoutlet",
+            ),
+        ];
+        for (to, via, expected) in cases {
+            let res = CreateCommand::parse_arg_to(&state, &to, Some(&via.to_string()))
+                .await
+                .unwrap();
+            assert_eq!(res, expected.to_string());
+        }
+
+        // "via" is ignored if "to" is passed as a full route
```

Comment:
Should it be ignored or raise an error? Because it's likely to be a configuration error in that case.

Reply:
Good point, I think it's better to return an error, yes.